llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -13062,13 +13127,39 @@ SDValue SITargetLowering::performXorCombine(SDNode *N,
   if (SDValue RV = reassociateScalarOps(N, DCI.DAG))
     return RV;
 
+  SelectionDAG &DAG = DCI.DAG;
+  EVT VT = N->getValueType(0);
   SDValue LHS = N->getOperand(0);
   SDValue RHS = N->getOperand(1);
 
+  if (VT == MVT::v2i32 && LHS.getNumOperands() > 1) {
```

Comment:
Add DAG comment explaining what this is doing. We probably should directly handle xor-as-fneg in selection of select